x-pack/solutions/observability/plugins/slo/server/services/purge_rollup_data.test.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { elasticsearchServiceMock } from '@kbn/core/server/mocks';
+import { Duration, DurationUnit } from '@kbn/slo-schema';
+import { createAPMTransactionErrorRateIndicator, createSLO } from './fixtures/slo';
+import { createSLORepositoryMock } from './mocks';
+import { SLORepository } from './slo_repository';
+import { BulkPurgeRollupData } from './bulk_purge_rollup_data';
+
+describe('purge rollup data', () => {
+  let mockRepository: jest.Mocked<SLORepository>;
+  let mockEsClient: jest.Mocked<ElasticsearchClient>;
+  let purgeRollupData: BulkPurgeRollupData;
+
+  beforeEach(() => {
+    mockRepository = createSLORepositoryMock();
+    mockEsClient = elasticsearchServiceMock.createElasticsearchClient();
+    purgeRollupData = new BulkPurgeRollupData(mockEsClient, mockRepository);
+  });
+
+  describe('happy path', () => {
+    it('successfully makes a query to remove SLI data older than 30 days', async () => {
+      const slo = createSLO({
+        id: 'test1',
+        indicator: createAPMTransactionErrorRateIndicator(),
```

Comment:
The SLO is set with an indicator, but for the purpose of this test we don't really care about it. We do care about its time window though. So it's better to explicitly set the time window of the SLO, and keep all the other irrelevant fields with their default values.


pseudo code:
```
const slo = createSLO({ id: "test1", timeWindow: sevenDaysRolling() })
await purgeRollupData.execute({
        list: ['test1'],
        purgePolicy: { purgeType: 'fixed_age', age: new Duration(30, DurationUnit.Day) },
      });
```